plugins/settings/settings_plugin.c

```diff
@@ -360,17 +610,158 @@ settings_plugin_dbus_handle_set_enabled(
+}
+
+static
+gboolean
+settings_plugin_dbus_handle_get_all_plugin_settings(
+    OrgSailfishosNfcSettings* iface,
+    GDBusMethodInvocation* call,
+    gpointer user_data)
+{
+    SettingsPlugin* self = THIS(user_data);
+
+    if (settings_plugin_access_allowed(self, call,
+        SETTINGS_ACTION_GET_ALL_PLUGIN_SETTINGS,
+        SETTINGS_DEFAULT_ACCESS_GET_ALL_PLUGIN_SETTINGS)) {
+        org_sailfishos_nfc_settings_complete_get_all_plugin_settings(iface,
+            call, settings_plugin_get_all_plugin_settings(self));
+    }
+    return TRUE;
+}
+
+static
+gboolean
+settings_plugin_dbus_handle_get_plugin_settings(
+    OrgSailfishosNfcSettings* iface,
+    GDBusMethodInvocation* call,
+    const char* plugin,
+    gpointer user_data)
+{
+    SettingsPlugin* self = THIS(user_data);
+
+    if (settings_plugin_access_allowed(self, call,
+        SETTINGS_ACTION_GET_PLUGIN_SETTINGS,
+        SETTINGS_DEFAULT_ACCESS_GET_PLUGIN_SETTINGS)) {
+        const SettingsPluginConfig* pc = g_hash_table_lookup(self->plugins,
+            plugin);
+
+        if (pc) {
+            org_sailfishos_nfc_settings_complete_get_plugin_settings(iface,
+              call, settings_plugin_config_variant(pc->config));
+        } else {
+            g_dbus_method_invocation_return_error_literal(call,
+                SETTINGS_DBUS_ERROR, SETTINGS_ERROR_UNKNOWN_PLUGIN, plugin);
+        }
+    }
+    return TRUE;
+}
+
+static
+gboolean
+settings_plugin_dbus_handle_get_plugin_value(
+    OrgSailfishosNfcSettings* iface,
+    GDBusMethodInvocation* call,
+    const char* plugin,
+    const char* key,
+    gpointer user_data)
+{
+    SettingsPlugin* self = THIS(user_data);
+
+    if (settings_plugin_access_allowed(self, call,
+        SETTINGS_ACTION_GET_PLUGIN_VALUE,
+        SETTINGS_DEFAULT_ACCESS_GET_PLUGIN_VALUE)) {
+        const SettingsPluginConfig* pc = g_hash_table_lookup(self->plugins,
+            plugin);
+
+        if (pc) {
+            GVariant* value = nfc_config_get_value(pc->config, key);
+
+            if (value) {
+                org_sailfishos_nfc_settings_complete_get_plugin_value(iface,
+                    call, g_variant_new_variant(value));
+                g_variant_unref(value);
+            } else {
+                /* What else could be wrong? */
+                g_dbus_method_invocation_return_error_literal(call,
+                    SETTINGS_DBUS_ERROR, SETTINGS_ERROR_UNKNOWN_KEY, key);
+            }
+        } else {
+            g_dbus_method_invocation_return_error_literal(call,
+                SETTINGS_DBUS_ERROR, SETTINGS_ERROR_UNKNOWN_PLUGIN, plugin);
+        }
+    }
+    return TRUE;
+}
+
+static
+gboolean
+settings_plugin_dbus_handle_set_plugin_value(
+    OrgSailfishosNfcSettings* iface,
+    GDBusMethodInvocation* call,
+    const char* plugin,
+    const char* key,
+    GVariant* var,
+    gpointer user_data)
+{
+    SettingsPlugin* self = THIS(user_data);
+
+    if (settings_plugin_access_allowed1(self, call,
```

Comment:
also here: not allowed -> silently ignored?

Reply:
If `settings_plugin_access_allowed*` functions/macros return FALSE, they complete the call with error. This behavior is described near the place where those functions/macros are defined but perhaps that comment should be replicated where they are actually being used. It's not an obvious behavior, indeed.

Reply:
Added comments.